### PR TITLE
updated unpaved function.  Is based on surface now.

### PR DESCRIPTION
**Pull request by gknisely**
@dgearhart @dnesbitt61 @kevinkreiser

### CODE REVIEW

File: src/baldr/directededge.cc [hunk 1]
```diff
@@ -66,7 +66,14 @@ bool DirectedEdge::destonly() const {
...
-  return false;  //TODO Fix.
+
+  Surface edge_surface = this->surface();
+  if (edge_surface == Surface::kPavedSmooth ||
+      edge_surface == Surface::kPaved ||
+      edge_surface == Surface::kPavedRough)
+    return false;
+
+  return true;
 }
```
Comment (dnesbitt61) [inline src/baldr/directededge.cc]: can't this be: 
return surface < = Surface::kPavedRough

Comment (dnesbitt61) [inline src/baldr/directededge.cc]: I mean:  return (surface() <= Surface::kPavedRough);  
simple! no temp variables or if statements

**Comment by gknisely** [comment]
agree...hold

**Comment by gknisely** [comment]
actually...should be this as you said you wanted cobblestone included as paved.
return (surface() <= Surface::kCompacted);

**Pull request merged**